File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        map<int,int> mp1;
        map<int,vector<int>> mp2;
        vector<int> v;
        for(int i:nums){
            mp1[i]++;
        }
        for(auto i:mp1){
            mp2[i.second].push_back(i.first);
        }
        for(auto it = mp2.rbegin(); it != mp2.rend() && k > 0; ++it){

            for(int num : it->second){

                v.push_back(num);
                k--;

                if(k == 0)
                    break;
            }
        }
        return v;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d = {}
        l = []
        for i in nums:
            d[i]=d.get(i,0)+1
        new = list(dict(sorted(d.items(),key = lambda x: x[1],reverse = True)).keys())
        for i in range(k):
            l.append(new[i])
        return l